src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -795,3 +817,25 @@ def is_cli_using_msal_auth():
             continue
         return i > j
     return len(v1.split(".")) == len(v2.split("."))
+
+
+def get_metadata(arm_endpoint, api_version="2022-09-01"):
+    metadata_url_suffix = f"/metadata/endpoints?api-version={api_version}"
+    try:
+        error_msg_fmt = "Unable to get metadata endpoints from the cloud.\n{}"
+        import requests
+        session = requests.Session()
+        metadata_endpoint = arm_endpoint + metadata_url_suffix
+        print(f"Retrieving ARM metadata from: {metadata_endpoint}")
+        response = session.get(metadata_endpoint)
+        if response.status_code == 200:
+            return response.json()
+        else:
+            msg = "Server returned status code {} for {}".format(response.status_code, metadata_endpoint)
+            raise AzureResponseError(error_msg_fmt.format(msg))
```

Comment:
this [error type](https://github.com/Azure/azure-cli/blob/fdb9aa742b404a433f207b573eea37341769020c/doc/error_handling_guidelines.md?plain=1#L33) directly comes under UserFault category. this should have separate handling based on error response code. For any ARM failures, please use the exception handler already built here: https://github.com/Azure/azure-cli-extensions/blob/762daced825bdf76c02021db6559d51a8b7191eb/src/connectedk8s/azext_connectedk8s/_utils.py#L420